### ISSUE TITLE
use_frameworks!  and dependency subspecs

### ISSUE THREAD

**Issue by ryhor** [CocoaPods/CocoaPods#4190]
Hello Everyone,

In my sample project i have 2 targets *TestFW2* and *TestFW2Tests*.

I also have a Pod named 'iOS_CommonComponents':

```
iOS_CommonComponents:
-subspec: Controllers
--- dependency: MBProgressHUD
-subspec: Views
--- dependency: MBProgressHUD
```

Here is the podfile of the project

```
source 'https://github.com/CocoaPods/Specs.git'

platform :ios, '8.0'

use_frameworks!

target 'TestFW2', do
    pod 'iOS_CommonComponents/Controllers'
end

target 'TestFW2Tests' do
    pod 'iOS_CommonComponents/Views'
end
```

**Problem is**: neither of the 'CommonComponents' project builds with the following error
```
ld: framework not found MBProgressHUD
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

It is reasonable, since both *Pods-TestFW2* and *Pods-TestFW2Tests* are only searching for frameworks inside their subfolders (see screenshot below) and not finding anything.

![1 png 2015-09-16 17-57-07](https://cloud.githubusercontent.com/assets/5094985/9908471/7949ee46-5c9c-11e5-96f5-0ad568316a95.png)

Maybe somebody knows a workaround, or this is expected behaviour, or maybe I'm doing something wrong.

Thank you very much in advance,
Ryhor

**Comment by neonichu** [comment]
This looks like a problem with deduplication — could you try turning that off and report back?

**Comment by ryhor** [comment]
Hi Boris, thanks very much for your input,

**1.** I can report that **disabling deduplication** indeed **helps** to battle the problem.
Each target gets its own set of frameworks. All links well.
![2](https://cloud.githubusercontent.com/assets/5094985/9953981/ebafcc7e-5ded-11e5-8fb5-0fcf609ad038.png)


**2.** Another workaround I created is adding this post-install hook to make the pods that are used in multiple targets search in the 'global' build folder (where non-duplicated go).

```
multiReferencePods = ['iOS_CommonComponents']
globalPath = '$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)'

post_install do |installer|
    installer.pods_project.targets.each do |target|
        multiReferencePods.each do |mrPodName|
            if target.name.end_with? mrPodName
                target.build_configurations.each do |config|
                    config.build_settings['FRAMEWORK_SEARCH_PATHS'] ||= ['$(inherited)', '$PODS_FRAMEWORK_BUILD_PATH', globalPath]
                end
            end
        end
    end
end
```

**Comment by Krizai** [comment]
Facing same issue
Workedarround with disabling deduplication, but really waiting for a fix for this.

**Comment by segiddins** [comment]
@mrackwitz would you mind looking into this?

**Comment by segiddins** [comment]
This ought to be fixed in the latest 1.0 betas, please let us know if it isn't.
